nexus/src/app/sagas/demo.rs

```diff
@@ -0,0 +1,126 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Demo saga, used for testing and interactive debugging
+//!
+//! The "Demo" saga exists so that developers and automated tests can create a
+//! saga that will not complete until they take some action to complete it.  The
+//! saga just waits until it gets the message that it should finish.  Users
+//! create demo sagas and complete them using requests to the internal API.
+//!
+//! The implementation is entirely in-memory, which means you have to send the
+//! completion message to the Nexus that's running the saga.  However, it does
+//! work across Nexus restarts, so this can be used to exercise the saga
+//! recovery path.
+//!
+//! It's tempting to build this only for development and not official releases,
+//! but that'd be more work, there's little downside to always including it, and
+//! it's conceivable that it'd be useful for production systems, too.
```

Comment:
:+1: agreed...I'd only advocate for removing it from production builds if leaving it in could cause damage, and it doesn't seem super dangerous.